syntax = "proto3";

package LlmModel;

import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";
import "UUID.proto";
import "User.proto";

service LlmModelController {
    rpc List(LlmModelListRequest) returns (stream LlmModel) {}
    rpc Create(LlmModel) returns (LlmModel) {}
    rpc Retrieve(LlmModelRetrieveRequest) returns (LlmModel) {}
    rpc Update(LlmModel) returns (LlmModel) {}
    rpc Destroy(LlmModel) returns (google.protobuf.Empty) {}
}

message LlmModel {
    UUID id = 1;
    google.protobuf.Timestamp created_at = 2;
    google.protobuf.Timestamp updated_at = 3;
    string llm_name = 4;
    string provider = 5;
    string model_version = 6;
    string api_key = 7;
    User user = 8;
}

message LlmModelListRequest {
}

message LlmModelRetrieveRequest {
    string id = 1;
}
